### PR TITLE
Move DocumentStorage Handling Out of Delta Manager

### LINKED ISSUE #6151: Remove Connect to Storage From DeltaManger, and Move creation flow to container
i investigated this, and it's not feasible without breaking the existing separation of concerns, as it would required getting the protocol handler into the delta manager. The existing abstraction is incorrect, as i don't think the containers storage should be controlled by delta manager at all, it's unclear why delta manger is involved in it's creation flow at all because it never even uses the storage

_Originally posted by @anthony-murphy in https://github.com/microsoft/FluidFramework/pull/6127#discussion_r632702203_

### PR DESCRIPTION
**Pull request by anthony-murphy**
fixes #6151

### CODE REVIEW

File: packages/loader/container-loader/src/retriableDocumentStorageService.ts [hunk 1]
```diff
@@ -34,9 +35,8 @@ export class RetriableDocumentStorageService implements IDocumentStorageService
         return this.internalStorageService.policies;
     }
 
-    public dispose() {
-        this.disposed = true;
-    }
+    public get disposed() {return this.runDisposedOnce.evaluated;}
+    public readonly dispose = ()=>this.runDisposedOnce.value;
```
Comment (vladsud) [inline packages/loader/container-loader/src/retriableDocumentStorageService.ts:39]: I find earlier logic easier to follow (and cheaper :))

**Review by vladsud — APPROVED** [review]

File: packages/loader/container-loader/src/retriableDocumentStorageService.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/loader/container-loader/src/retriableDocumentStorageService.ts]: reverted

**Review by ChumpChief — APPROVED** [review]

File: packages/loader/container-loader/src/container.ts [hunk 2]
```diff
@@ -393,8 +397,8 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     private _attachState = AttachState.Detached;
 
     // Active chaincode and associated runtime
-    private _storageService: IDocumentStorageService | undefined;
-    private get storageService() {
+    private _storageService: IDocumentStorageService & IDisposable | undefined;
```
Comment (ChumpChief) [inline packages/loader/container-loader/src/container.ts:400]: Github ate my comment :-/

Do you think `IDocumentStorageService` should extend `IDisposable`?  Or alternatively, is there benefit to this typing as opposed to just typing it as `RetriableDocumentStorageService` like `DeltaManager` does today?

Comment (anthony-murphy) [reply packages/loader/container-loader/src/container.ts:400]: There is already a bug for it, and yes i do: #5980 . My change is a bit pre-emptive, in #5902 i'll wrap `RetriableDocumentStorageService`  with the protocol tree handler

Comment (znewton) [reply packages/loader/container-loader/src/container.ts:400]: Started the work for this change in #8083

**Pull request merged**